setup.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+time=`date '+%Y%m%d_%H%M'`
+VAG_COMMON_DIR="/vagrant/"
+LOG_DIR="$VAG_COMMON_DIR/log"
+OUTFILE="$LOG_DIR/${time}_vagrant-out.log"
+
+runSetup() {
+    step=1
+
+    if [ ! -d $LOG_DIR ]; then
+        mkdir $LOG_DIR
+    fi
+
+    if [ ! -e $OUTFILE ]; then
+        touch $OUTFILE
+    fi
+
+    echo " "
+    echo " "
+    echo "==> Running Provision Script: $0:"
+    echo " "
+    echo "$((step++)). Updating apt-git repositories..."
+    sudo apt-get update  >> $OUTFILE 2>&1 
+    [ $? -ne 0 ] && exit 1
+    sudo apt-get -yq install software-properties-common >> $OUTFILE 2>&1 
+    [ $? -ne 0 ] && exit 1
+    echo " "
+
+    echo "$((step++)). Installing Python & PIP..."
+    curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py  >> $OUTFILE 2>&1 
+    [ $? -ne 0 ] && exit 1
+
+    /usr/bin/python3 get-pip.py  >> $OUTFILE 2>&1 
+    [ $? -ne 0 ] && exit 1
+
+    sudo DEBIAN_FRONTEND=noninteractive apt-get -yq install python3-dev python3-pip  >> $OUTFILE 2>&1 
+    [ $? -ne 0 ] && exit 1
```

Comment:
why are you trying to get pip manually and then installing it with package manager (apt) afterwards?
If you are -yq options, why are you also using DEBIAN_FRONTEND=noninteractive?

Reply:
Noted. Upgraded OS version and removed these lines from setup.sh accordingly.
Please see latest commit.